src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java

```diff
@@ -39,25 +41,35 @@
  * <p>
  * This plugin keeps track of its version through a {@link VersionNumberAction} attached
  * to the project.  Each build that uses this plugin has its own VersionNumberAction,
- * and this contains the builds today/this month/ this year/ all time.  When incrementing
+ * and this contains the builds today / this month / this year / all time.  When incrementing
  * each of these values, unless they're overridden in the configuration the value from
  * the previous build will be used.
  * </p>
+ * <p>
+ * Such a value can be either overridden with a plain number or with an environment-variable.
+ * In the later case the value will be read from the environment-variable at build-time. If
+ * it cannot be parsed as an integer the value from the previous build will be incremented
+ * and used instead.
+ * </p>
  *
  * @author Carl Lischeske - NETFLIX
+ * @author Deniz Bahadir - BENOCS
  */
 public class VersionNumberBuilder extends BuildWrapper {
     
     private static final DateFormat defaultDateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    // Pattern:   ${VAR_NAME} or $VAR_NAME
+    private static final String envVarPattern = new String("^(?:\\$\\{(\\w+)\\})|(?:\\$(\\w+))$");
+
     private final String versionNumberString;
     private final Date projectStartDate;
     private final String environmentVariableName;
     private final String environmentPrefixVariable;
     
-    private int oBuildsToday;
-    private int oBuildsThisMonth;
-    private int oBuildsThisYear;
-    private int oBuildsAllTime;
+    private String oBuildsToday;
+    private String oBuildsThisMonth;
+    private String oBuildsThisYear;
+    private String oBuildsAllTime;
```

Comment:
Yes, I agree. XStream is not aware about types of simple objects. My apologies for a mis-leading comment.